File: php/examples/subscribe-example.php
<?php

    ## Capture Publish and Subscribe Keys from Command Line
    $publish_key   = isset($argv[1]) ? $argv[1] : false;
    $subscribe_key = isset($argv[2]) ? $argv[2] : false;
    $channel_name  = isset($argv[3]) ? $argv[3] : 'hello_world2';

    # Print usage if missing info.
    if (!($publish_key && $subscribe_key)) {
echo("
    ==============
    EXAMPLE USAGE:
    ==============
    php ./subscribe-example.php PUBLISH-KEY SUBSCRIBE-KEY

");
        exit();
    }

    ## Require Pubnub API
    echo("Loading Pubnub.php Class\n");
    require('../Pubnub.php');

    ## -----------------------------------------
    ## Create Pubnub Client API (INITIALIZATION)
    ## -----------------------------------------
    echo("Creating new Pubnub Client API\n");
    $pubnub = new Pubnub( $publish_key, $subscribe_key );

    ## ------------------------------
    ## Listen for Message (SUBSCRIBE)
    ## ------------------------------
    echo("Listening for Messages (Press ^C to stop)\n");
    $pubnub->subscribe(array(
        'channel'  => $channel_name,        ## REQUIRED Channel to Listen
        'callback' => function($message) {  ## REQUIRED Callback With Response
            var_dump($message);  ## Print Message
            usleep(100);
            return true;         ## Keep listening (return false to stop)
        }
    ));

?>

File: ruby/examples/subscribe-example.rb
##including required libraries
require 'rubygems'
require './lib/pubnub.rb'

##declaring publish_key, subscribe_key, secret_key, cipher_key
publish_key   = 'demo'
subscribe_key = 'demo'
secret_key    = 'demo'
cipher_key    = 'demo'
ssl_on        = !!ARGV[2]


## Print usage if missing info.
if !subscribe_key
    puts('
    Get API Keys at http://www.pubnub.com/account
    ==============
    EXAMPLE USAGE:
    ==============
    ruby subscribe-example.rb PUBLISH-KEY SUBSCRIBE-KEY SSL-ON
    ruby subscribe-example.rb demo demo true
    ')  
end

## Create Pubnub Client API (INITIALIZATION)

puts('Creating new Pubnub Client API')
pubnub = Pubnub.new(publish_key,subscribe_key,secret_key,cipher_key,ssl_on=false)

## Listen for Messages (SUBSCRIBE) 

puts('Listening for new messages with subscribe() Function')
puts('Press CTRL+C to quit.')

pubnub.subscribe({
    'channel'  => 'HelloWorld',
    'callback' => lambda do |message|
        puts(message) ## print message
        return true   ## keep listening?
    end
})
